### ISSUE TITLE
Headphones: pressing volume buttons unmutes internal speaker

### ISSUE THREAD

**Issue by benni092** [rg35xx-cfw/Koriki#50]
### What happened?

If you plug in your headphones, the audio is routed to the headphones and the internal speaker gets muted. As soon as you press one of the volume buttons, the internal speaker gets unmuted. Then the audio comes from both, speaker and headphones. If you plug out and in your headphones, the speaker gets muted again.

Can anyone tell me the location of the file that controlls the audio/volume so i can try to fix it?

bw.

**Comment by acmeplus** [comment]
Yes, it's a known issue, the audio is not managed via configuration file but with a binary that needs to be fixed. The program that manages de audio does not take into account whether the headphones are plugged in, so when audio changes it enables the speaker again.

**Comment by benni092** [comment]
The file
/sys/devices/virtual/switch/h2w/state
gives me 0 if headphones are unplugged and 1 if they are plugged.

The file in
/etc/triggerhappy/triggers.d/multimedia_keys.conf controls the screenbrightness?

And the file
/etc/asound.conf can control the audio?

If i add a script to the triggerhappy file that mutes the speaker every time the headphones are plugged in and the volume buttons are pressed can this work?

**Comment by benni092** [comment]
[apps.zip](https://github.com/rg35xx-cfw/Koriki/files/12329304/apps.zip)

I have written a very quick and dirty script that watches the state of the headphone switch and than mutes the speaker every 0.5 second if a headphone is detected.

Extract the files to /system/.simplemenu/apps on the SHARE partition.
On your RG35xx navigate to apps&games --> apps and run the Headphones&.sh file.

You must rerun the Script every boot.

I wish all people that love the RG35xx a nice day!

**Comment by benni092** [comment]
Unfortunaly the new Koriki 1.0 dont mute the speaker at all. I plug in headphones, but audio came from both speaker and headphones.

**Comment by benni092** [comment]
Same on Koriki 1.0.1 but when i start an emu, volume goes to 100% until i press volume down. Then it becomes the volume level that was set before.
I have checked the file in /sys/devices/virtual/switch/h2w/state. It constantly gives me 0.
I tried this with a Koriki 0.9.1 card, and the file becomes 1 as it should. So hardware is functioning.

**Comment by grandsion** [comment]
Depuis la version 1.0.1 , l'insertion du casque audio dans le port jack ne coupe plus les haut parleur de la console.
